src/services/spotify.service.ts

```diff
@@ -0,0 +1,78 @@
+
+const BASE_URL = 'https://api.spotify.com/v1';
+
+export class SpotifyService {
+  private _accessToken: string | null = null;
+
+  constructor() {
+    let parsed = new URLSearchParams(window.location.search);
+    this._accessToken = parsed.get("access_token");
+    
+    if (!this._accessToken) {
+      console.error('YIKES BOI, No access_token query param found');
+    }
+  }
+
+  getAccessToken(): string | null {
+    return this._accessToken;
+  }
+
+  async fetchMe() {
+    return await fetch(`${BASE_URL}/me`, {
```

Comment:
So these all return the promise returned by `await fetch`. Not sure how familiar you are with promises, but in a super basic sense its an object that we can call `.next((data) => { ...do something with data...})` where that inside method only fires after the data has been fetched

Reply:
I am familiar with promises, but I tend to get bogged down in the syntax of `.next((data) => {something})` with all of the parentheses, curly brackets, resolves, etc.. 

The alternative syntax of `async await` lets you write asynchronous code in a way that's more similar to regular ol' synchronous JS. It's basically like [training wheels](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await) for newbies (like me) when working with promises

Reply:
Yeah async await can be useful, I usually stick to promises so you can have both async and sync code in the same method.

If you really want to learn about how to do async code in the best way check out https://rxjs-dev.firebaseapp.com/guide/overview . This is a lib for using `observables`, they kinda dumb complicated